lib/modules/manager/regex/readme.md

```diff
@@ -13,21 +13,24 @@ The first two required fields are `fileMatch` and `matchStrings`. `fileMatch` wo
 In order for Renovate to look up a dependency and decide about updates, it then needs the following information about each dependency:
 
 - The dependency's name
-- Which `datasource` to look up (e.g. npm, Docker, GitHub tags, etc)
-- Which version scheme to apply (defaults to `semver`, but also may be other values like `pep440`)
+- Which [`datasource`](https://docs.renovatebot.com/modules/datasource/#supported-datasources) to look up (e.g. [npm](https://docs.renovatebot.com/modules/datasource/#npm-datasource), [Docker](https://docs.renovatebot.com/modules/datasource/#docker-datasource), [GitHub tags](https://docs.renovatebot.com/modules/datasource/#github-tags-datasource), [etc.](https://docs.renovatebot.com/modules/datasource/#supported-datasources))
+- Which [version scheme](https://docs.renovatebot.com/modules/versioning/#supported-versioning) to apply (defaults to [`semver`](https://docs.renovatebot.com/modules/versioning/#semantic-versioning), but also may be other values like [`pep440`](https://docs.renovatebot.com/modules/versioning/#pep440-versioning))
+
+Note: capture groups below are similar to [available fields](https://docs.renovatebot.com/templates/#other-available-fields), but these available fields are not available in templates below, because most of those are coming from managers, and `regex` is a manager.
 
 Configuration-wise, it works like this:
 
 - You must capture the `currentValue` of the dependency in a named capture group
-- You must have either a `depName` capture group or a `depNameTemplate` config field
-- You can optionally have a `packageName` capture group or a `packageNameTemplate` if it differs from `depName`
-- You must have either a `datasource` capture group or a `datasourceTemplate` config field
-- You can optionally have a `depType` capture group or a `depTypeTemplate` config field
-- You can optionally have a `versioning` capture group or a `versioningTemplate` config field. If neither are present, `semver` will be used as the default
-- You can optionally have an `extractVersion` capture group or an `extractVersionTemplate` config field
-- You can optionally have a `currentDigest` capture group.
-- You can optionally have a `registryUrl` capture group or a `registryUrlTemplate` config field
-  - If it's a valid URL, it will be converted to the `registryUrls` field as a single-length array.
+- You must have either a `depName` capture group or a [`depNameTemplate` config field](https://docs.renovatebot.com/configuration-options/#depnametemplate)
```

Comment:
I haven't looked into how to do a self-link yet without hardcoding an absolute URL.

Reply:
We can use _relative_ links, but I always forget how that works. 🙈 

@viceice can you help here?

Reply:
should work, sample:
https://github.com/renovatebot/renovate/blob/41fd7fbf46ca9e3a46f5ace2a1de213929d9e347/docs/usage/getting-started/private-packages.md#L344

Reply:
I'm not sure these relatives are a good idea. 
![image](https://user-images.githubusercontent.com/2906988/163029000-7aa8818d-8d77-482d-b1e5-7952ca75666a.png)

All we get is warnings upon warnings (IntelliJ IDEA):
![image](https://user-images.githubusercontent.com/2906988/163029134-e8af2eda-fc25-43ed-8baa-355d488ac873.png)

We also lose context, e.g. `../../datasource/` assumes that this part of the code is actually hosted at the url `/modules/manager/regex/` and hides the fact that it's referring to `/modules/datasource`. I came at this from the maintenance perspective. If there's any refactor of docs, it'll be hard to update, or even discover that it's linked.

The linked "GitLab Runner" example links to the same folder `/docs` in source and in prod, but the links I added here pretty much link to virtual folders in prod (i.e. configuration-options).

How about going with absolute (i.e. host-relative) links?
```suggestion
- You must have either a `depName` capture group or a [`depNameTemplate` config field](/configuration-options/#depnametemplate)
```

Reply:
Whatever we do, it should be easy and simple to maintain, and we shouldn't end up with an IDE filled with warnings.

I don't think we can use _absolute_ links though. Quote from the [manual for MkDocs on linking to pages](https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages):

> Warning
> 
> Using absolute paths with links is not officially supported. Relative paths are adjusted by MkDocs to ensure they are always relative to the page. Absolute paths are not modified at all. This means that your links using absolute paths might work fine in your local environment but they might break once you deploy them to your production server.

Reply:
the problem is probably that we shift the modules pages around before building the final docs.

so i would suggest that we use repo relative links. and when copying files we fix the links to new relationship.

Reply:
@MaronHatoum I think I would be happy to merge, but we reached a stalemate here, it's really not in scope of this PR to fix how MkDocs, but I don't know what to do, how to link to those other parts of the docs in a way that would make this PR approved.

Reply:
> when copying files we fix the links to new relationship.

@viceice to me this sounds like there needs to be some tooling improvements before we can merge? otherwise we end up with broken links.

Reply:
use absolute urls for now, as we don't have any tooling yet to fix the relative paths for publishing